app/src/test/java/ai/elimu/analytics/util/DateHelperTest.kt

```
@@ -0,0 +1,129 @@
+        val parsedDate = DateHelper.eventDateFormat.parse(dateString)
+        val expectedDate = SimpleDateFormat("yyyy-MM-dd", Locale.US).parse(dateString)
+        assertEquals("eventDateFormat should parse date correctly", expectedDate, parsedDate)
+    }
+
+    @Test
+    fun testEventDateFormat_handlesInvalidDateFormat() {
+        val invalidDateString = "2025-13-06" // Invalid month
+        val parsedDate = try {
+            DateHelper.eventDateFormat.parse(invalidDateString)
+            null
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)